internal/dag/httpproxy_processor.go

```diff
@@ -171,9 +204,40 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 				}
 				svhost.DownstreamValidation = dv
 			}
-		} else if tls.ClientValidation != nil {
-			sw.SetInvalid("Spec.VirtualHost.TLS passthrough cannot be combined with tls.clientValidation")
-			return
+
+			if proxy.Spec.VirtualHost.AuthorizationConfigured() {
+				auth := proxy.Spec.VirtualHost.Authorization
+				ref := defaultExtensionRef(auth.ServiceRef)
+
+				if ref.APIVersion != projcontourv1alpha1.GroupVersion.String() {
+					sw.SetInvalid("Spec.Virtualhost.Authorization.ExtensionServiceRef specifies an unsupported resource version %q",
+						auth.ServiceRef.APIVersion)
+					return
+				}
+
+				if ref.Kind != "ExtensionService" {
```

Comment:
The way it's done in the spec is the recommended way for doing cross-kind resources in CRDs now. It also gives us the space later to support other Kinds if we want to, with the defaults allowing us to make it so you don't have to specify everything for now. If we don't do something like this now, it will be difficult to add later because of API guarantees.

Reply:
What is difficult about adding fields later on? The API guarantee is fine with adding fields.

Reply:
I understand these are patterns in the community, but at the same time, I'm reluctant of taking on that complexity in the event we "may someday need it".

Reply:
I don't think we can remove the version, but sure, we can take the Kind out for now, and add it back if we need it.